bocs_control/data_reader.py

```diff
@@ -0,0 +1,84 @@
+"""=============================================================================
+DataReader CLASS
+--------------------------------------------------------------------------------
+
+============================================================================="""
+from datetime import datetime as dt
+import logging
+import sys
+import threading
+
+import serial
+from serial.tools.list_ports import comports
+
+# ===============================================================================
+class DataReader(threading.Thread):
+    """
+    DataReader describes a thread whose purpose is to read lines of serial input
+    from a specified serial port and enqueue said lines of serial input into a
+    shared FIFO queue.
+    """
+
+    def __init__(self, name, port_name, shared_queue):
+        threading.Thread.__init__(self)
+        logging.info(f"Initialising serial reader on port name {port_name}")
+        self.name = name
+        self.port_name = port_name
+        self.queue = shared_queue
+
+        try:
+            logging.info(f"Checking port {self.port_name} availability")
+            self.check_port_available()
+        except serial.serialutil.SerialException as ex:
+            logging.error(f"Port {self.port_name} is unavailable")
+            raise RuntimeError() from ex
+
+        try:
+            logging.info(f"Opening port {self.port_name} for read")
```

Comment:
Should the `logger` call be inside the `try`?

Reply:
Changed.